### PR TITLE
LinQ reenumeration fixes

### LINKED ISSUE #2040: 🐛LINQ reenumeration issues - worse performance, possibility of corrupted data
**:bug: Describe the bug**
LINQ queries applied on IEnumerable interface are not properly materialized once, but reenumerated by several materializing calls of First, FirstOrDefault, Last etc.... Chain of linq statements must be executed once data should be provided to next method or returned. Current situation leads to worse performance because of re-executing aplied statements and also can lead to corrupted data.

Avoid using First or silimar action repeatedly in one block of code.
Replace Any + First by FirstOrDefault

**To Reproduce**
Check all LINQ queries and their flow. Check all methods that returns IEnumerable and use better interface forcing developer to execute query before return (like IList, ICollection)

For instance
image
Data are not copied once this method is called, but later once it is materialized. If "targetWellbore" is changed before that, it will leads to corrupted data

--------

Several calls of First on Objects properties of "source" and "sourceQuery"
image

--------

This:
image

can be simplified to : 
image

### PR DESCRIPTION
**Pull request by petrlicman**
## Type of change

[//]: # (Mark any of the types of change that apply.)

* [x] Bugfix
* [ ] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [ ] Frontend
* [x] API
* [ ] WITSML
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Query/ObjectQueries.cs [hunk 1]
```diff
@@ -53,16 +52,18 @@ public static IWitsmlObjectList GetWitsmlObjectsByType(EntityType type)
...
+            o.Uid = string.Empty;
+            o.NameWell = string.Empty;
+            o.NameWellbore = string.Empty;
+            o.Name = string.Empty;
             if (objectProperty != null)
             {
                 o = QueryHelper.AddPropertyToObject(o, objectProperty, objectPropertyValue);
             };
+
+            // TODO: REMOVE CASTING!
```
Comment (jaroslavbliznak) [inline Src/WitsmlExplorer.Api/Query/ObjectQueries.cs:66]: Will this be remove in another task?

File: Src/WitsmlExplorer.Api/appsettings.Development.json [hunk 2]
```diff
@@ -1,4 +1,8 @@
 {
+"MongoDb": {
```
Comment (jaroslavbliznak) [inline Src/WitsmlExplorer.Api/appsettings.Development.json]: this should not be part of the commit

File: Src/WitsmlExplorer.Api/HttpHandlers/JobHandler.cs [hunk 3]
```diff
@@ -75,7 +75,7 @@ private static bool IsAdminOrDeveloper(string token)
         {
             JwtSecurityTokenHandler handler = new();
             JwtSecurityToken jwt = handler.ReadJwtToken(token);
-            IEnumerable<string> userRoles = jwt.Claims.Where(n => n.Type == "roles").Select(n => n.Value);
+            IEnumerable<string> userRoles = jwt.Claims.Where(n => n.Type == "roles").Select(n => n.Value).ToList();
```
Comment (jaroslavbliznak) [inline Src/WitsmlExplorer.Api/HttpHandlers/JobHandler.cs]: Add List<string>

File: Src/WitsmlExplorer.Api/Query/ObjectQueries.cs (hunk 1 above)
Comment (petrlicman) [reply Src/WitsmlExplorer.Api/Query/ObjectQueries.cs:66]: Yes, it must be and will be removed in future, it is note that this casting is not save and should not be used in this way

File: Src/WitsmlExplorer.Api/HttpHandlers/JobHandler.cs (hunk 3 above)
Comment (petrlicman) [reply Src/WitsmlExplorer.Api/HttpHandlers/JobHandler.cs]: Changed to List type

File: Src/WitsmlExplorer.Api/appsettings.Development.json (hunk 2 above)
Comment (petrlicman) [reply Src/WitsmlExplorer.Api/appsettings.Development.json]: this file has been reverted to original version

**Review by jaroslavbliznak — APPROVED** [review]
